src/PHPCR/Shell/Console/Command/Phpcr/NodeListCommand.php

```diff
@@ -100,12 +100,13 @@ public function execute(InputInterface $input, OutputInterface $output): int
             if (!$globHelper->isGlobbed($session->getAbsPath($path))) {
                 throw $e;
             }
+            $path = $session->getAbsPath($path);
```

Comment:
that last failure is confusing. @dantleech @alexander-schranz can you help me here?

the phpcr-shell seems to handle a pwd (PhpcrSession class). the problem is that the NodeListCommand calls phpcr-utils PathHelper::getParentPath with potentially relative paths instead of first making them absolute. and the PathHelper::getParentPath did not sanitize the input to require leading `/` - even though it documents "Get the parent path of a valid absolute path.".

this test is passing `num*` as path. according to 3v4l, `substr('afds', 0, false)` results in an empty string. and the previous test does `numberPropertyNode/jcr:con*` which just happens to work accidentally because it contains a `/`.

the correct thing to do imo would be to make the path absolute first. do you agree? @dantleech  i am not very familiar with the shell, would appreciate if you can check around this line to see if i mess up something.

Reply:
fixed with ae6ff43d8f0139800a683e542316d488a1328a94